content/300-guides/025-migrate/500-data-migration.mdx

```diff
@@ -0,0 +1,260 @@
+- `prisma migrate dev` is only run against development database 
+- The expanding and contracting steps are handled in separate branches
+
+For this guide, you will modify the following schema by replacing the `published` boolean field with a `status` enum:
+
+```prisma file=prisma/schema.prisma
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model Post {
+  id        Int      @id @default(autoincrement())
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+  title     String
+  content   String?
+  published Boolean  @default(false)
+  viewCount Int      @default(0)
+}
+```
+
+## Expand your schema with a new column
+
+Checkout to a new branch from your `main` branch:
+
+```bash
+git checkout -b create-status-field
+```
+
+Make the following updates to your Prisma schema:
+
+- Create a `Status` enum with the following values: `Unknown`, `Draft`, `InReview`, and `Published`
+- Add a `status` column to the `Post` model
+- Mark the `published` field as optional
+
+```prisma file=prisma/schema.prisma highlight=12,13,18-24;edit
+model Post {
+  id        Int      @id @default(autoincrement())
+  title     String
+  content   String?
+  published Boolean? @default(false)
+  status    Status
+}
+
+enum Status {
+  Unknown
+  Draft
+  InProgress
+  InReview
+  Published
+}
+```
+
+Create a new migration to sync the Prisma schema with the database schema:
+
+```bash
+npx prisma migrate dev --name add-status-column
+```
+
+Prisma Migrate will give you the following warning because the field being added to the database is non-nullable, and the database contains existing data which require a default value.
+
+<cmdResult>
+
+<cmd>
+
+```bash no-copy
+Prisma schema loaded from prisma/schema.prisma
+Datasource "db": PostgreSQL database "data-migration", schema "public" at "localhost:5401"
+
+Error:
+⚠️ We found changes that cannot be executed:
+
+  • Step 1 Added the required column `status` to the `Post` table without a default value. There are 4 rows in this table, it is not possible to execute this step.
+
+You can use prisma migrate dev --create-only to create the migration file, and manually modify it to address the underlying issue(s).
+Then run prisma migrate dev to apply it and verify it works.
+```
+
+</cmd>
+</cmdResult>
+
+Exit from the migration step and update the schema by adding a default value for the `status` field by adding the `@default()` attribute function.
+
+```prisma file=prisma/schema.prisma highlight=13;edit
+model Post {
+  id        Int      @id @default(autoincrement())
+  title     String
+  content   String?
+  published Boolean? @default(false)
+  status    Status   @default(Unknown)
+}
+
+enum Status {
+  Unknown
+  Draft
+  InProgress
+  InReview
+  Published
+}
+```
+
+Re-run `prisma migrate dev` to generate and apply a migration to the database.
```

Comment:
Let's put the command into its own `terminal` snippet to be consistent and make it harder to overread